### ISSUE TITLE
Unclear about config-precedence

### ISSUE THREAD

**Issue by cideM** [prettier/prettier#3432]
I skipped the template because it did not seem to apply to this problem. I am using `vim-prettier` but my problem seems to related to the prettier CLI. If I run prettier from within neovim the command that's ran is:

`prettier --print-width 90 --tab-width 4 --use-tabs false --semi false --single-quote true --bracket-spacing true --jsx-bracket-same-line false --trailing-comma all --parser flow --config-precedence prefer-file --stdin-filepath /path/to/file.js --stdin`

This is based on my settings in `nvim.init`. I don't have a configuration file in the project root, but I would expect prettier to now fall back to the CLI options I just passed. Instead it uses the default options. Changing the `config_precedence` in `nvim.init` to `file-override`, I get this command:

`prettier --print-width 90 --tab-width 4 --use-tabs false --semi false --single-quote true --bracket-spacing true --jsx-bracket-same-line false --trailing-comma all --parser flow --config-precedence file-override  --stdin-filepath /path/to/file.js --stdin` 

and the CLI options are used.

Looking at the documentation, it is also not clear to me what the difference between both `config-precedence` options really is.

> file-override: config file take precedence over CLI options
> prefer-file: If a config file is found will evaluate it and ignore other CLI options. If no config file is found CLI options will evaluate as normal.

This sounds like the same behavior, in different words.

If this is a user error, or a problem of `vim-prettier` let me know. If you need  more information or a better bug report to reproduce, let me know as well.

**Comment by azz** [comment]
file-override means the CLI flags will be merged with your configuration file. With prefer-file however, if a configuration file is found, all CLI flags are ignored.

**Comment by duailibe** [comment]
I was able to reproduce with a clean folder, no config file, no config on package.json, and `prefer-file` ignored my CLI flags.

**Comment by azz** [comment]
Hmm, there's tests covering that case...

**Comment by duailibe** [comment]
Just to double check I'm not doing something wrong:

```
› ls -a 
./  ../  .git/  .gitignore  node_modules/  package.json  test.js  yarn.lock
```

package.json:
```json
{
  "dependencies": {
    "prettier": "1.9.1"
  }
}
```

```› prettier --no-semi --config-precedence prefer-file test.js
foo();
foo();
foo();
foo();
foo();
```

**Comment by azz** [comment]
Looking like a bug. Can you run with `--loglevel=debug`?

**Comment by duailibe** [comment]
I always forget about `--loglevel`

```
[debug] normalized argv: {"_":["test.js"],"bracket-spacing":false,"color":true,"debug-check":false,"debug-print-doc":false,"editorconfig":true,"flow-parser":false,"insert-pragma":false,"jsx-bracket-same-line":false,"list-different":false,"require-pragma":false,"semi":false,"single-quote":false,"stdin":false,"support-info":false,"use-tabs":false,"version":false,"with-node-modules":false,"write":false,"config-precedence":"prefer-file","loglevel":"debug","ignore-path":".prettierignore"}
[debug] resolve config from 'test.js'
[debug] loaded options `{}`
[debug] applied config-precedence (prefer-file): {"filepath":"test.js"}
```

**Comment by azz** [comment]
Confirmed my suspicion, regression from the editorconfig PR. loaded options should be `null`, not `{}`.

**Comment by duailibe** [comment]
Yep, running with `--no-editorconfig` works

**Comment by cideM** [comment]
Wow this, uhm, progressed quickly. Thanks everyone for dealing with this so quickly 😍

**Comment by mitermayer** [comment]
Just for reference, this caused this issue (https://github.com/prettier/vim-prettier/issues/84) and the suggested fix of `--no-editorconfig` seemed to fix this. Will do a new release of vim-prettier which disables editorconfig and then can re-enable it again after a new prettier release that contains this fix

**Comment by mitermayer** [comment]
also this one https://github.com/prettier/vim-prettier/issues/83

**Comment by duailibe** [comment]
Fixed in 1.9.2!
